src/components/SearchBar.js

```diff
@@ -0,0 +1,114 @@
+import styled from 'styled-components';
+import { AiOutlineSearch } from 'react-icons/ai';
+
+function SearchBar({ getRegion, getCountry }) {
+    const debounce = (func, timeout = 300) => {
+        let timerId;
+
+        return (...args) => {
+            if (timerId) {
+                clearTimeout(timerId);
+            }
+
+            timerId = setTimeout(() => {
+                func(...args);
+            }, timeout);
+        };
+    };
+
+    const setRegion = (e) => {
+        getRegion(e.target.value);
+    };
+
+    const setCountry = debounce((e) => {
+        getCountry(e.target.value);
+    });
+
+    return (
+        <Wrapper>
+            <PositioningElement>
+                <Input
+                    type='search'
+                    placeholder='Search for a country…'
+                    onChange={setCountry}
+                />
+                <SearchIconStyled />
+            </PositioningElement>
+            <Select
+                onChange={setRegion}
+                aria-label='Search countries by region'
+            >
+                <option value=''>Filter By Region</option>
+                <option value='africa'>Africa</option>
+                <option value='america'>America</option>
+                <option value='asia'>Asia</option>
+                <option value='europe'>Europe</option>
+                <option value='oceania'>Oceania</option>
+            </Select>
+        </Wrapper>
+    );
+}
+
+const Wrapper = styled.div`
+    align-items: center;
+    display: flex;
+    flex-wrap: wrap;
+    font-size: 0.875rem;
+    font-weight: var(--fontWeight-400);
+    gap: 40px;
+    justify-content: space-between;
+    margin-top: clamp(1.5rem, 0.97rem + 2.3vw, 3rem);
+    padding-inline: clamp(1rem, -0.45rem + 6vw, 5rem);
+`;
+
+const PositioningElement = styled.div`
+    border-radius: 0.3125em;
```

Comment:
I would avoid numbers that seem random, like this one.

Maybe you want to look into making a theme object with pre-defined spacings and colours.

Example: https://tailwindcss.com/docs/customizing-spacing